browser/ui/android/brave_rewards/rewards_page_activity_helper.cc

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/android/brave_rewards/rewards_page_activity_helper.h"
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "brave/build/android/jni_headers/RewardsPageActivity_jni.h"
+
+namespace brave_rewards {
+
+void OpenURLForRewardsPageActivity(const std::string& url) {
```

Comment:
I'm confused about what is going on here, why do we need to call out to java to open a new tab?

Reply:
When the rewards page is opened in an Android webview, calls to `window.open` or links with `target=_blank` will open links in the current webview, instead of a new (regular) tab. Since we want those links opened in regular tabs, we have the front-end dispatch to its page handler, and eventually to this helper.

Reply:
we have the similar implementation to open tab in native client for Leo, wallet, etc.

Reply:
what I'm saying is that we can open tabs from cpp so why are we doing this in java? The method this calls opens the url as a singleton tab, is that necessary here? If it is, I am assuming there is likely some existing upstream functionality for that similar to SingletonTabs on desktop. I asked in chromium slack

Reply:
If this doesn't need to be a singleton then we do not need java here. aichat actually delegates from java to cpp to open urls in a new tab. We should only be (maybe) using that java code if we need the `openNewOrSelectExistingTab` functionality

```
content::OpenURLParams params(
      ConversationUrl(conversation->get_conversation_uuid()),
      content::Referrer(), WindowOpenDisposition::NEW_FOREGROUND_TAB,
      ui::PAGE_TRANSITION_FROM_API, false);
  web_contents->OpenURL(params, {});
```

Reply:
we agreed to keep the implementation with using TabUtils function directly.